hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/lock/LockManager.java

```diff
@@ -64,10 +69,14 @@ public void lock() {
       boolean acquired = false;
       while (retryCount <= maxRetries) {
         try {
+          metrics.startLockApiTimerContext();
           acquired = lockProvider.tryLock(writeConfig.getLockAcquireWaitTimeoutInMs(), TimeUnit.MILLISECONDS);
           if (acquired) {
+            metrics.updateLockAcquiredMetric();
+            metrics.startLockHeldTimerContext();
             break;
           }
+          metrics.updateLockNotAcquiredMetric();
```

Comment:
shouldn't we move this to catch block ?

Reply:
Moved it to catch as well. But can't remove it from here as tryLock can simply return false for some implementations and not throw an exception.